### PR TITLE
Frontend testing setup

### PR DESCRIPTION
**Pull request by Vucis**

### CODE REVIEW

File: frontend/tests/utils/api.ts
Comment (Gerwoud) [inline frontend/tests/utils/api.ts:1]: Empty file?

**Review by JibrilExe — APPROVED** [review]

File: frontend/package.json [hunk 1]
```diff
@@ -8,8 +8,10 @@
     "build": "tsc && vite build",
     "lint": "eslint . --ext ts,tsx --report-unused-disable-directives --max-warnings 0",
     "preview": "vite preview --port 5173",
-    "cypress:test": "cypress run",
-    "test": "npm run cypress:test"
+    "cy:test": "cypress run",
+    "cy:open": "cypress open",
+    "coverage": "vitest run --coverage",
+    "test": "vitest run"
```
Comment (AronBuzogany) [inline frontend/package.json:14]: Make sure that `npm run test` runs all tests, component and e2e

I think you can add `vitest run && npm run cy:test` for this.

File: frontend/package.json [hunk 2]
```diff
@@ -46,7 +48,8 @@
     "@typescript-eslint/eslint-plugin": "^6.21.0",
     "@typescript-eslint/parser": "^6.21.0",
     "@vitejs/plugin-react": "^4.2.1",
-    "cypress": "^13.7.0",
+    "@vitest/coverage-v8": "^1.5.2",
+    "cypress": "^13.8.1",
```
Comment (AronBuzogany) [inline frontend/package.json:52]: These are dev dependencies

File: frontend/package.json [hunk 3]
```diff
@@ -55,9 +58,14 @@
     "i18next": "^23.10.1",
     "i18next-browser-languagedetector": "^7.2.1",
     "i18next-http-backend": "^2.5.0",
+    "jest": "^29.7.0",
+    "jsdom": "^24.0.0",
+    "npm-run-all": "^4.1.5",
```
Comment (AronBuzogany) [inline frontend/package.json:63]: Same here

File: frontend/package.json [hunk 4]
```diff
@@ -55,9 +58,14 @@
...
     "react-i18next": "^14.1.0",
     "scheduler": "^0.23.0",
-    "typescript": "^5.2.2",
-    "vite": "^5.1.7"
+    "ts-jest": "^29.1.2",
```
Comment (AronBuzogany) [inline frontend/package.json:66]: And here

**Review by AronBuzogany — CHANGES_REQUESTED** [review]

File: frontend/package.json (hunk 2 above)
Comment (Vucis) [reply frontend/package.json:52]: They are in dev dependencies?

File: frontend/package.json (hunk 3 above)
Comment (Vucis) [reply frontend/package.json:63]: Same here?

File: frontend/package.json (hunk 2 above)
Comment (Vucis) [reply frontend/package.json:52]: Or should they not be dev dependencies?

File: frontend/package.json (hunk 1 above)
Comment (Vucis) [reply frontend/package.json]: [ee1992f](https://github.com/SELab-2/UGent-3/pull/290/commits/ee1992fbd9fab84640246cd8651974235b2a52fa)

**Review by AronBuzogany — APPROVED** [review]

**Pull request merged**